### PR TITLE
Fix: using custom path to TestFramework

### LINKED ISSUE #217: Custom path to Test Framework not used in 0.8.1
| Question    | Answer
| ------------| ---------------
| Infection version | 0.8.1 (`infection.phar --version`)

Patch to come.

### PR DESCRIPTION
**Pull request by tomtomau**
This PR:

- [x] Covered by tests

Fixes https://github.com/infection/infection/issues/217

* As a user, I want to use a custom path to the TestFramework (fixes #217)
* As a user, I want a descriptive error message if my custom path does not exist, rather than use a fallback

### CODE REVIEW

File: tests/Finder/TestFrameworkFinderTest.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+use Infection\Finder\Exception\FinderException;
+use Infection\Finder\SourceFilesFinder;
+use Infection\Finder\TestFrameworkFinder;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Finder\Finder;
+
+class TestFrameworkFinderTest extends TestCase
+{
+    public function test_it_can_load_a_custom_path(): void
```
Comment (BackEndTea) [inline tests/Finder/TestFrameworkFinderTest.php]: Please remove the `: void`  since that breaks test on 7.0

**Comment by BackEndTea** [comment]
👍 Its better to throw an error rather than using a fallback if the user specified a location.

I'd like to extend this behavior to other places, where the user specifies locations of files.

File: tests/Finder/TestFrameworkFinderTest.php (hunk 1 above)
Comment (tomtomau) [reply tests/Finder/TestFrameworkFinderTest.php]: ah of course, habit :)

File: tests/Finder/TestFrameworkFinderTest.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+use Infection\Finder\Exception\FinderException;
+use Infection\Finder\TestFrameworkFinder;
+use PHPUnit\Framework\TestCase;
+
+class TestFrameworkFinderTest extends TestCase
+{
+    public function test_it_can_load_a_custom_path()
+    {
+        $filename = '/tmp/infection-test-framework-finder-test' . uniqid('', false);
```
Comment (sidz) [inline tests/Finder/TestFrameworkFinderTest.php]: this test won't work on windows I think. Could you please use `sys_get_temp_dir` instead? also it'll be a good idea to clean up.

you can use symfony filesystem class for example. We already have couple tests which do similar things.

**Comment by tomtomau** [comment]
Updated @sidz

File: tests/Finder/TestFrameworkFinderTest.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+declare(strict_types=1);
+
+namespace Infection\Tests\Finder;
+
+use Infection\Finder\Exception\FinderException;
+use Infection\Finder\TestFrameworkFinder;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Filesystem\Filesystem;
+
+class TestFrameworkFinderTest extends TestCase
```
Comment (borNfreee) [inline tests/Finder/TestFrameworkFinderTest.php]: Please, use setUp and tearDown methods to create and remove temp folders:

https://github.com/infection/infection/blob/12bbf5ab87d840c8b64ec973de5e1506d27087b5/tests/TestFramework/PhpUnit/Config/Builder/InitialConfigBuilderTest.php#L49-L64

File: tests/Finder/TestFrameworkFinderTest.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+{
+    public function test_it_can_load_a_custom_path()
+    {
+        $fileSystem = new Filesystem();
+        $filename = $fileSystem->tempnam(sys_get_temp_dir(), 'infection-test');
+        $fileSystem->touch([$filename]);
+
+        $frameworkFinder = new TestFrameworkFinder('not-used', $filename);
+
+        static::assertEquals($filename, $frameworkFinder->find(), 'Should return the custom path');
```
Comment (borNfreee) [inline tests/Finder/TestFrameworkFinderTest.php]: let's use `$this->assertEquals()` to be consistent across the project

**Comment by tomtomau** [comment]
Updated! @borNfreee

**Comment by djmattyg007** [comment]
Would you be able to create a new release (ie 0.8.2) that includes this patch?

**Comment by borNfreee** [comment]
@djmattyg007 I will try to release it today or tomorrow

**Comment by 1ed** [comment]
Also affected by this. @borNfreee could you please release a new version if you have some time? Thanks!

**Comment by borNfreee** [comment]
@djmattyg007 @1ed 

Released under [0.8.2](https://github.com/infection/infection/releases/tag/0.8.2)

Sorry for the delay

**Comment by 1ed** [comment]
That's grate! Thank you very much @borNfreee! There will be a phar too?

**Comment by borNfreee** [comment]
> There will be a phar too?

@1ed sorry for this delay, uploaded the PHAR

**Pull request merged**